File: LearningManagementBackend/src/main/java/org/learn/learningmanagementbackend/dto/response/NotificationResponse.java
package org.learn.learningmanagementbackend.dto.response;

import lombok.Getter;
import lombok.Setter;
import org.learn.learningmanagementbackend.enums.NotificationType;

import java.time.LocalDateTime;

@Getter
@Setter
public class NotificationResponse {
    private Integer id;
    private Integer userId;
    private String userName;
    private String title;
    private String body;
    private NotificationType type;
    private Boolean isRead;
    private LocalDateTime createdAt;
}

File: learning_management_app/lib/models/admin/notification_response.dart
class NotificationResponse {
  final int? id;
  final int? userId;
  final String? userName;
  final String? title;
  final String? body;
  final String? type;
  final bool? isRead;
  final String? createdAt;

  NotificationResponse({
    this.id,
    this.userId,
    this.userName,
    this.title,
    this.body,
    this.type,
    this.isRead,
    this.createdAt,
  });

  factory NotificationResponse.fromJson(Map<String, dynamic> json) {
    return NotificationResponse(
      id: json['id'],
      userId: json['userId'],
      userName: json['userName'],
      title: json['title'],
      body: json['body'],
      type: json['type'],
      isRead: json['isRead'],
      createdAt: json['createdAt'],
    );
  }
}
